tools/generators/legacy/src/Generator/ConsolidateTargets.swift

```diff
@@ -354,7 +354,7 @@ extension ConsolidatedTarget {
         self.targets = targets
         let aTarget = self.targets.first!.value
 
-        name = aTarget.name
+        self.name = aTarget.product.moduleName ?? aTarget.name
```

Comment:
Since each target could have different module names (because of `select()`), how should we handle that? I think we have three options:

1. They can't be consolidated
2. We fall back to the target name
3. We do like what you have, just using the first (sorted)

I don't like 1, but 3 or 2 seem fine to me?

Reply:
I think we should use the target name as a fallback ( option 2 ). So if the module name is unique we use it, if not we use the target and if that doesn't help we use the label.

Reply:
Can you make that change? You would collect all of the module names in this init, and if all are non-nil and the same, use it, otherwise use the target.name like before.

Reply:
I implemented this way, based on fixtures works as expected. The implementation is a bit ugly, if you see how to improve it let me know.